cpp/src/join/mixed_join_common_utils.cuh

```diff
@@ -160,6 +161,39 @@ struct pair_expression_equality : public expression_equality<has_nulls> {
   }
 };
 
+/**
+ * @brief Equality comparator that composes two row_equality comparators.
+ */
+struct double_row_equality_comparator {
+  row_equality const equality_comparator;
+  row_equality const conditional_comparator;
+
+  __device__ bool operator()(size_type lhs_row_index, size_type rhs_row_index) const noexcept
+  {
+    using experimental::row::lhs_index_type;
+    using experimental::row::rhs_index_type;
+
+    return equality_comparator(lhs_index_type{lhs_row_index}, rhs_index_type{rhs_row_index}) &&
+           conditional_comparator(lhs_index_type{lhs_row_index}, rhs_index_type{rhs_row_index});
+  }
+};
+
+// A CUDA Cooperative Group of 4 threads for the hash set.
+auto constexpr DEFAULT_MIXED_SEMI_JOIN_CG_SIZE = 1;
```

Comment:
I might not be reading this correctly.  Does the size in the comment (4) contradict that specified in the constant definition (1)?

Reply:
Actually, should this not be set to `4` instead of `1`?

Reply:
Recently updated `cg_size` from 4 to 1. Thanks for pointing out. Will update the comment as well.